### DISCUSSION TITLE
3. 네트워크 - 네트워크 계층

### DISCUSSION THREAD

**Discussion by seongeun42** [Shortudy-10th/tech-for-developer#157] [질의 응답]
네트워크 계층 질의응답

**Comment by zini9188** [comment]
## Q. 네트워크 계층의 역할은 무엇인가요?

중간 라우터를 통한 라우팅을 포함하여 패킷 포워딩을 담당하는 계층입니다.

**Comment by zini9188** [comment]
## Q. NAT에 대해 설명해주세요.
ip 패킷의 tcp/ip 포트 숫자와 소스 및 목적지의 ip 주소 등을 재기록하며 라우터를 통해 네트워크 트래픽을 주고 받는 기술을 말합니다.

**Reply by zini9188** [reply]
## Q. NAT을 사용하는 이유는 무엇인가요?
공인 ip 주소를 절약할 수 있으며 방화벽 설정을 통해 트래픽을 거르고 ip를 숨겨 보안성이 향상됩니다.

**Comment by seongeun42** [comment]
## Q. 네트워크 계층의 중요한 기능으로는 포워딩과 라우팅이 있습니다. 포워딩과 라우팅에 대해 설명해주세요.

포워딩은 라우터의 입력 포트부터 출력 포트까지 패킷이 이동하는 것이고, 매우 짧은 시간의 단위를 갖기 때문에 하드웨어에서 실행됩니다. 라우팅은 네트워크 전체에서 출발지부터 목적지까지의 경로를 결정하는 것으로, 포워딩보다 긴 시간 동안 이루어지며 보통 소프트웨어에서 실행됩니다.

**Comment by seongeun42** [comment]
## Q. IP에 대해 설명해주세요.

송신 호스트와 수신 호스트가 패킷 교환 네트워크에서 정보를 주고받는 데 사용하는 정보 위주의 프로토콜로, 비신뢰성과 비연결성의 특징을 가집니다.

**Comment by seongeun42** [comment]
## Q. IP주소란 무엇인가요?

네트워크 상에서 장치들이 서로를 식별하고 통신하기 위해 사용하는 특수한 번호입니다.

**Comment by seongeun42** [comment]
## Q. IPv4와 IPv6에 대해 설명해주세요.

IPv4는 주소가 32비트로 구성되어 있다. 컴퓨터뿐 아니라 스마트폰 등의 다양한 장치들이 나오며 32비트로 모든 장치의 주소를 표현할 수 없게 되자, 128비트 주소 체계를 가지는 IPv6가 나오게 되었습니다.

**Comment by seongeun42** [comment]
## Q. 서브넷이란 무엇인가요?

IP주소에서 네트워크 영역을 부분적으로 나눈 부분 네트워크를 의미합니다. 서브넷을 사용함으로써 라우터가 특정 장치를 찾을 때 범위를 효율적으로 줄여나가며 찾을 수 있습니다.

**Comment by seongeun42** [comment]
## Q. ICMP에 대해 간단히 설명해주세요.

ICMP는 TCP/IP에서 IP패킷을 처리할 때 발생되는 문제를 알려주는 프로토콜입니다. 패킷의 목적지 호스트가 없는 등의 에러 상황에 에러가 발생했다는 걸 송신자에게 알려주기 위해 에러 메시지를 전달합니다.

**Comment by seongeun42** [comment]
## Q. DHCP에 대해 간단히 설명해주세요.

DHCP는 네트워크 장치에 자동적으로 IP 주소, 서브넷 마스크, 게이트웨이 및 DNS 정보를 할당하는 프로토콜입니다.

**Comment by LireEruel** [comment]
## Q. IP 주소와 서브넷 마스크의 차이는 무엇인가요?

IP 주소는 네트워크 상에서 호스트 또는 장치를 식별하는 데 사용되는 고유한 숫자입니다.
서브넷 마스크는 IP 주소의 네트워크 부분과 호스트 부분을 분리하는데 사용되는 숫자입니다.
서브넷 마스크를 사용하여 IP 주소를 서브넷으로 분리하면 네트워크 관리와 라우팅을 효과적으로 할 수 있습니다.

**Reply by LireEruel** [reply]
## Q. 서브넷으로 분리하면 네트워크 관리와 라우팅을 효과적으로 할 수 있는 이유가 뭔가요

서브넷으로 IP 네트워크를 분리했을 때 2가지 장점이 있습니다.

첫번째로  트래픽 분리입니다.
서브넷을 사용하면 네트워크 내에서 트래픽을 분리할 수 있습니다. 각 서브넷은 자체적인 논리적인 네트워크로 간주되어 서브넷 간의 트래픽은 라우터를 통해 라우팅되어 분리됩니다. 이렇게 하면 논리적인 분리와 보안을 유지할 수 있습니다.

두번째로 IP 주소 관리입니다.
서브넷을 나누면 각 서브넷에 할당된 IP 주소 범위가 줄어들어 IP 주소 공간의 낭비를 방지하고, 주소 할당을 관리하기 쉬워집니다.
